src/plugins/timeframe-selector-plugin.ts

```diff
@@ -265,133 +289,240 @@ export class TimeframeSelectorPlugin extends UIPlugin<TimeframeSelectorPluginSty
         this.shouldRender = true;
     }
 
-    setData(data: FlameChartNodes) {
-        this.data = data;
+    makeFlameChartDots() {
+        if (this.flameChartNodes) {
+            const flameChartDots: Dot[] = [];
+            const tree = flatTree(this.flameChartNodes);
+            const { min, max } = getFlatTreeMinMax(tree);
+
+            this.min = min;
+            this.max = max;
+
+            this.clusters = metaClusterizeFlatTree(tree, () => true);
+            this.actualClusters = clusterizeFlatTree(
+                this.clusters,
+                this.renderEngine.zoom,
+                this.min,
+                this.max,
+                2,
```

Comment:
maybe we could add a rule for magic numbers
looking at the PR, I need to go to the clusterizeFlatTree implementation to see what 2 is
maybe having it as a constant?
I am also fine with this, It is just me being used more to descriptive constants and less to magic numbers